packages/dialog/src/provider/use-dialog-hook.ts

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+import { UiDialogsControllerContext } from '.';
+
+type DialogActions = {
+  openDialog: () => void;
+  closeDialog: () => void;
+};
+
+type UseDialogHook = {
+  isOpen: boolean;
+  actions: DialogActions;
+};
+
+export const useDialog = (dialogId: string): UseDialogHook => {
+  const dialogController = React.useContext(UiDialogsControllerContext);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const actions = React.useMemo(() => {
+    return {
+      openDialog: () => dialogController.openDialog(dialogId),
+      closeDialog: () => dialogController.closeDialog(dialogId),
+    };
+  }, [dialogId]);
+
+  React.useEffect(() => {
+    let opened = false;
+
+    dialogController.openedDialogs.map((dialog) => {
+      if (dialog === dialogId) {
+        opened = true;
+      }
+    });
+
+    setIsOpen(opened);
+  });
+
+  return { isOpen, actions };
+};
```

Comment:
This is the hook that will be used inside each component and will return the functions that each instance of a dialog can use to open or close.